### PR TITLE
Support Two-legged OAuth flow

### PR DESCRIPTION
**Pull request by mantas-sidlauskas**
**What changed?**
Adding generic, client-credentials OAuth 2 support.

**Why?**
There is self-signed JWT provider already implemented. Adding client credentials flow support. 

**How did you test it?**
Added unit test. 

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: src/main/java/com/uber/cadence/internal/compatibility/proto/serviceclient/GrpcServiceStubs.java [hunk 1]
```diff
@@ -71,6 +73,9 @@ final class GrpcServiceStubs implements IGrpcServiceStubs {
   private static final Metadata.Key<String> RPC_ENCODING_HEADER_KEY =
       Metadata.Key.of("rpc-encoding", Metadata.ASCII_STRING_MARSHALLER);
 
+  private static final Metadata.Key<String> AUTHORIZATION_HEADER_KEY =
+      Metadata.Key.of("cadence-authorization", Metadata.ASCII_STRING_MARSHALLER);
```
[the remark this answers is not in the document]
Comment (mantas-sidlauskas) [reply src/main/java/com/uber/cadence/internal/compatibility/proto/serviceclient/GrpcServiceStubs.java:77]: Yes, that's correct. But unfortunately, server expects `cadence-authorization` https://github.com/uber/cadence/blob/fcbc00ec11ed0af72fae91ac1922ca9b2e87594b/common/rpc.go#L51

But yes, we need to go with "authorization" header eventually

**Comment by shijiesheng** [comment]
It seems that only OAuth is only supported in GRPC. Are we planning to make changes to Tchannel as well?

**Comment by mantas-sidlauskas** [comment]
> It seems that OAuth is only supported in GRPC. Are we planning to make changes to Tchannel as well?

@shijiesheng  TChannel has OAuth [capabilities already](https://github.com/uber/cadence-java-client/blob/master/src/main/java/com/uber/cadence/serviceclient/WorkflowServiceTChannel.java#L246). In this PR I'm adding gRPC OAuth support as well

**Review by 3vilhamster — APPROVED** [review]

**Comment by natemort** [comment]
Hey @mantas-sidlauskas , can you bump the oauth client version to avoid https://www.cve.org/CVERecord?id=CVE-2021-22573 ? I'm happy to merge once that's addressed.

**Pull request merged**